### DISCUSSION TITLE
Is frontend and backend on one server not bad ?

### DISCUSSION THREAD

**Discussion by pullmann4rent** [remix-run/remix#6232] [Q&A] [answered]
Hello,

on Remix frontend and backend are togehter, but it is not bad when both is in one server ? Much people tend to split it, to have clientside a server and backend a server. So how makes remix here a difference or a advantage ?

**Comment by sergiodxa** [comment, marked as answer]
It's not bad, it's actually how the web has worked forever and how a lot of apps (like GitHub) works.

Also just because Remix has a browser and a server part doesn't mean you have to have the frontend and backend together.

You can build your backend as an API with any tech you prefer, then use Remix as a client fetching the API from the loaders and actions.

The main reason to keep Remix as both frontend and backend is that now you can have direct access to your DB, this means you can query exactly the data your UI will need, instead of fetching one or more generic endpoints on an API, and it will be faster since you don't need to go throw the network, just do a DB query and a few ms later you have the data.
